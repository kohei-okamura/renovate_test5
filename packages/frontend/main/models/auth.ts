/*
 * Copyright © 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { Permission } from '@zinger/enums/lib/permission'
import { Staff } from '~/models/staff'

/**
 * 認証・認可情報.
 */
export type Auth = Readonly<{
  /** システム管理者フラグ */
  isSystemAdmin: boolean

  /** 権限 */
  permissions: Permission[]

  /** スタッフ */
  staff: Staff
}>
