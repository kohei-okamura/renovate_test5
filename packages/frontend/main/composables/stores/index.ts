/*
 * Copyright Â© 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { createSessionStore } from '~/composables/stores/create-session-store'
import { NuxtContext } from '~/models/nuxt'

type CreateGlobalStoreParams = Pick<NuxtContext, '$api'>

export const createGlobalStore = (params: CreateGlobalStoreParams) => ({
  session: createSessionStore(params)
})

export type GlobalStore = ReturnType<typeof createGlobalStore>
