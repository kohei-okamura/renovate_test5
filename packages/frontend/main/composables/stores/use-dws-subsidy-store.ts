/*
 * Copyright Â© 2021 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { InjectionKey, reactive } from '@nuxtjs/composition-api'
import { assign } from '@zinger/helpers'
import { createStore } from '~/composables/stores/utils'
import { usePlugins } from '~/composables/use-plugins'
import { UserDwsSubsidy } from '~/models/user-dws-subsidy'
import { DwsSubsidiesApi } from '~/services/api/dws-subsidies-api'

export const createDwsSubsidyState = () => ({
  dwsSubsidy: undefined as UserDwsSubsidy | undefined
})

export function useDwsSubsidyStore () {
  const { $api } = usePlugins()
  const state = reactive(createDwsSubsidyState())
  const actions = {
    async get (params: DwsSubsidiesApi.GetParams) {
      assign(state, await $api.dwsSubsidies.get(params))
    },
    async update (params: Parameters<typeof $api.dwsSubsidies.update>[0]) {
      assign(state, await $api.dwsSubsidies.update(params))
    }
  }
  return createStore({ actions, state })
}

export type DwsSubsidyData = ReturnType<typeof createDwsSubsidyState>

export type DwsSubsidyStore = ReturnType<typeof useDwsSubsidyStore>

export type DwsSubsidyState = DwsSubsidyStore['state']

export const dwsSubsidyStoreKey: InjectionKey<DwsSubsidyStore> = Symbol('dwsSubsidyStore')

export const dwsSubsidyStateKey: InjectionKey<DwsSubsidyState> = Symbol('dwsSubsidyState')
