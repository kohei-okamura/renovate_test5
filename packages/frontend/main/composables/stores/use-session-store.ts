/*
 * Copyright Â© 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { InjectionKey } from '@nuxtjs/composition-api'
import { SessionState, SessionStore } from '~/composables/stores/create-session-store'
import { usePlugins } from '~/composables/use-plugins'

export function useSessionStore () {
  const { $globalStore } = usePlugins()
  return $globalStore.session
}

export const sessionStateKey: InjectionKey<SessionState> = Symbol('sessionState')

export const sessionStoreKey: InjectionKey<SessionStore> = Symbol('sessionStore')
