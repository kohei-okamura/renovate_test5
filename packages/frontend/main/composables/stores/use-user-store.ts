/*
 * Copyright Â© 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { InjectionKey, reactive } from '@nuxtjs/composition-api'
import { assign } from '@zinger/helpers'
import { createStore } from '~/composables/stores/utils'
import { usePlugins } from '~/composables/use-plugins'
import { BankAccount } from '~/models/bank-account'
import { Contract } from '~/models/contract'
import { DwsCertification } from '~/models/dws-certification'
import { DwsProject } from '~/models/dws-project'
import { LtcsInsCard } from '~/models/ltcs-ins-card'
import { LtcsProject } from '~/models/ltcs-project'
import { User } from '~/models/user'
import { UserDwsCalcSpec } from '~/models/user-dws-calc-spec'
import { UserDwsSubsidy } from '~/models/user-dws-subsidy'
import { UserLtcsCalcSpec } from '~/models/user-ltcs-calc-spec'
import { UserLtcsSubsidy } from '~/models/user-ltcs-subsidy'
import { UsersApi } from '~/services/api/users-api'

export const createUserState = () => ({
  bankAccount: undefined as BankAccount | undefined,
  contracts: [] as Contract[],
  dwsCalcSpecs: [] as UserDwsCalcSpec[],
  dwsCertifications: [] as DwsCertification[],
  dwsProjects: [] as DwsProject[],
  dwsSubsidies: [] as UserDwsSubsidy[],
  ltcsCalcSpecs: [] as UserLtcsCalcSpec[],
  ltcsInsCards: [] as LtcsInsCard[],
  ltcsProjects: [] as LtcsProject[],
  ltcsSubsidies: [] as UserLtcsSubsidy[],
  user: undefined as User | undefined
})

export function useUserStore () {
  const { $api } = usePlugins()
  const state = reactive(createUserState())
  const actions = {
    async get (params: UsersApi.GetParams) {
      assign(state, await $api.users.get(params))
    },
    async update ({ form, id }: Parameters<typeof $api.users.update>[0]) {
      assign(state, await $api.users.update({ form, id }))
    },
    async updateBankAccount ({ form }: Parameters<typeof $api.bankAccounts.update>[0]) {
      assign(state, await $api.bankAccounts.update({ form }))
    }
  }
  return createStore({ actions, state })
}

export type UserData = ReturnType<typeof createUserState>

export type UserStore = ReturnType<typeof useUserStore>

export type UserState = UserStore['state']

export const userStoreKey: InjectionKey<UserStore> = Symbol('userStore')

export const userStateKey: InjectionKey<UserState> = Symbol('userState')
