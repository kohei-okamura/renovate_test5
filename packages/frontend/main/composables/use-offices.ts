/*
 * Copyright Â© 2021 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { computed, reactive, toRefs, watch } from '@nuxtjs/composition-api'
import { OfficeQualification } from '@zinger/enums/lib/office-qualification'
import { Permission } from '@zinger/enums/lib/permission'
import { Purpose } from '@zinger/enums/lib/purpose'
import { usePlugins } from '~/composables/use-plugins'
import { OfficeId } from '~/models/office'
import { UserId } from '~/models/user'
import { VSelectOption } from '~/models/vuetify'
import { RefOrValues, unref } from '~/support/reactive'

type BaseParams = {
  isCommunityGeneralSupportCenter?: boolean
  permission?: Permission
  qualifications?: OfficeQualification[]
  userId?: UserId
}
type ParamsWithPurpose = BaseParams & {
  internal?: never
  purpose?: Exclude<Purpose, typeof Purpose.unknown>
}
type ParamsWithInternal = BaseParams & {
  internal: true
  purpose?: never
}
type Params = RefOrValues<ParamsWithPurpose | ParamsWithInternal>

export type OfficeOption = VSelectOption<OfficeId> & {
  keyword: string
}

const createOfficeResolver = (xs: VSelectOption<OfficeId>[]) => {
  return (id: OfficeId | undefined, alternative = '-') => (id && xs.find(x => x.value === id)?.text) ?? alternative
}

export const useOffices = (params: Params) => {
  const { $api } = usePlugins()
  const state = reactive({
    isLoadingOffices: true,
    officeOptions: [] as OfficeOption[]
  })
  const resolveOfficeAbbr = computed(() => createOfficeResolver(state.officeOptions))
  watch(
    computed(() => ({
      isCommunityGeneralSupportCenter: unref(params.isCommunityGeneralSupportCenter),
      internal: unref(params.internal),
      permission: unref(params.permission),
      purpose: unref(params.purpose),
      qualifications: unref(params.qualifications),
      userId: unref(params.userId)
    })),
    async ({ internal, purpose, ...rest }) => {
      state.isLoadingOffices = true
      state.officeOptions = await $api.options.offices({ ...rest, purpose: internal ? Purpose.internal : purpose })
      state.isLoadingOffices = false
    },
    { immediate: true }
  )
  return {
    ...toRefs(state),
    resolveOfficeAbbr
  }
}
