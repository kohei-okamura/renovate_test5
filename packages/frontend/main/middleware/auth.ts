/*
 * Copyright © 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { Context } from '@nuxt/types'
import { Permission } from '@zinger/enums/lib/permission'
import { useAuth } from '~/composables/use-auth'

/**
 * 認証・認可ミドルウェア.
 */

type AuthMiddleware = {
  (...permissions: Permission[]): (context: Context) => Promise<void>
}

export const auth: AuthMiddleware = (...permissions) => async ({ app, redirect, route }: Context) => {
  const { session } = app.$globalStore
  const { prepare, isAuthenticated, isAuthorized } = useAuth(session)
  const redirectPromise = (path: string) => new Promise<void>(() => redirect(path))
  await prepare()
  if (!isAuthenticated.value) {
    // 未認証の場合はトップページにリダイレクトする
    // 画面を表示させないため下記の Promise は resolve しないこと
    return redirectPromise(`/?path=${route.fullPath}`)
  } else if (!isAuthorized.value(permissions)) {
    // 認可に失敗（＝必要な権限がない）場合は NotFound とする
    // 画面を表示させないため下記の Promise は resolve しないこと
    return redirectPromise('/notfound')
  } else {
    // 認証・認可に成功した場合はなにもしない
    return Promise.resolve()
  }
}
