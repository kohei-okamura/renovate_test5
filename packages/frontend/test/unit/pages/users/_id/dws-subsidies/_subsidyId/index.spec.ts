/*
 * Copyright © 2021 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { MountOptions, Wrapper } from '@vue/test-utils'
import { Permission } from '@zinger/enums/lib/permission'
import { UserDwsSubsidyType } from '@zinger/enums/lib/user-dws-subsidy-type'
import { mocked } from '@zinger/helpers/testing/mocked'
import Vue from 'vue'
import { dwsSubsidyStateKey } from '~/composables/stores/use-dws-subsidy-store'
import { sessionStoreKey } from '~/composables/stores/use-session-store'
import { userStateKey } from '~/composables/stores/use-user-store'
import { useOffices } from '~/composables/use-offices'
import { Auth } from '~/models/auth'
import DwsSubsidyViewPage from '~/pages/users/_id/dws-subsidies/_subsidyId/index.vue'
import { createDwsSubsidyResponseStub } from '~~/stubs/create-dws-subsidy-response-stub'
import { createDwsSubsidyStoreStub } from '~~/stubs/create-dws-subsidy-store-stub'
import { DWS_SUBSIDY_ID_MIN } from '~~/stubs/create-dws-subsidy-stub'
import { createUseOfficesStub } from '~~/stubs/create-use-offices-stub'
import { createUserResponseStub } from '~~/stubs/create-user-response-stub'
import { createUserStoreStub } from '~~/stubs/create-user-store-stub'
import { createAuthStub } from '~~/test/helpers/create-auth-stub'
import { provides } from '~~/test/helpers/provides'
import { setupComponentTest } from '~~/test/helpers/setup-component-test'

jest.mock('~/composables/use-offices')

describe('pages/users/_id/dws-subsidies/_subsidyId/index.vue', () => {
  const { mount } = setupComponentTest()
  const userResponse = createUserResponseStub()
  const userStore = createUserStoreStub(userResponse)

  let wrapper: Wrapper<Vue>

  type MountComponentParams = {
    auth?: Partial<Auth>
    options?: MountOptions<Vue>
  }

  function mountComponent (params: MountComponentParams = {}, subsidyType?: UserDwsSubsidyType) {
    const dwsSubsidyResponse = createDwsSubsidyResponseStub(DWS_SUBSIDY_ID_MIN, subsidyType)
    const dwsSubsidyStore = createDwsSubsidyStoreStub(dwsSubsidyResponse)
    const auth = params.auth ?? { isSystemAdmin: true }
    const options = params.options ?? {}
    wrapper = mount(DwsSubsidyViewPage, {
      ...options,
      ...provides(
        [dwsSubsidyStateKey, dwsSubsidyStore.state],
        [sessionStoreKey, createAuthStub(auth)],
        [userStateKey, userStore.state]
      )
    })
  }

  function unmountComponent () {
    wrapper.destroy()
  }

  beforeAll(() => {
    mocked(useOffices).mockReturnValue(createUseOfficesStub())
  })

  afterAll(() => {
    mocked(useOffices).mockReset()
  })

  it('should be rendered correctly', () => {
    mountComponent()
    expect(wrapper).toMatchSnapshot()
    unmountComponent()
  })

  describe.each([
    ['給付率', UserDwsSubsidyType.benefitRate],
    ['定率負担', UserDwsSubsidyType.copayRate],
    ['給付額', UserDwsSubsidyType.benefitAmount],
    ['本人負担額', UserDwsSubsidyType.copayAmount]
  ])('z-data-card-item（自治体助成情報）', (label, subsidyType) => {
    const selector = '[data-data-card]'

    it(`should be rendered z-data-card-item（${label}） when userDwsSubsidyType is ${subsidyType}`, () => {
      mountComponent({}, subsidyType)
      expect(wrapper).toContainElement(selector)
      expect(wrapper.find(selector)).toMatchSnapshot()
      unmountComponent()
    })
  })

  describe('FAB', () => {
    const requiredPermissions: Permission[] = [Permission.updateUserDwsSubsidies]

    it('should be rendered when session auth is system admin', () => {
      mountComponent()
      expect(wrapper).toContainElement('[data-fab]')
      unmountComponent()
    })

    it(`should be rendered when the staff has permission: ${requiredPermissions}`, () => {
      const auth = {
        permissions: requiredPermissions
      }

      mountComponent({ auth })

      expect(wrapper).toContainElement('[data-fab]')
      unmountComponent()
    })

    it(`should not be rendered when the staff does not have permission: ${requiredPermissions}`, () => {
      const auth = {
        permissions: Permission.values.filter(x => !requiredPermissions.includes(x))
      }

      mountComponent({ auth })

      expect(wrapper).not.toContainElement('[data-fab]')
      unmountComponent()
    })
  })
})
