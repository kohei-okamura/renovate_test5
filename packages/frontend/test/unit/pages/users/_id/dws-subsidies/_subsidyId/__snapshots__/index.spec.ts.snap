// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/users/_id/dws-subsidies/_subsidyId/index.vue should be rendered correctly 1`] = `
<div
  breadcrumbs="[object Object],[object Object],[object Object]"
  class="z-page-stub"
  compact="compact"
>
  <div
    class="z-page-stub-slot-tabs"
  />
  <div
    class="z-page-stub-slot-title"
  />
  <z-user-summary-stub
    user="[object Object]"
  />
  <div
    class="z-data-card-stub"
    data-data-card=""
    title="自治体助成情報"
  >
    <div
      class="z-data-card-stub-slot-actions"
    />
    <z-data-card-item-stub
      icon="M19,19H5V8H19M16,1V3H8V1H6V3H5C3.89,3 3,3.89 3,5V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V5C21,3.89 20.1,3 19,3H18V1M17,12H12V17H17V12Z"
      label="適用期間"
      value="10"
    >
      <time
        datetime="1983-05-01"
      >
        昭和58年5月1日
      </time>
      <span>
        〜
      </span>
      <time
        datetime="2020-07-21"
      >
        令和2年7月21日
      </time>
    </z-data-card-item-stub>
    <z-data-card-item-stub
      icon="M15,23H13V21H15V23M19,21H17V23H19V21M15,17H13V19H15V17M7,21H5V23H7V21M7,17H5V19H7V17M19,17H17V19H19V17M15,13H13V15H15V13M19,13H17V15H19V13M21,9A2,2 0 0,1 23,11V23H21V11H11V23H9V15H3V23H1V15A2,2 0 0,1 3,13H9V11A2,2 0 0,1 11,9V7A2,2 0 0,1 13,5H15V1H17V5H19A2,2 0 0,1 21,7V9M19,9V7H13V9H19Z"
      label="助成自治体名"
      value="羽咋市"
    />
    <z-data-card-item-stub
      label="助成自治体番号"
      value="714798"
    />
    <z-data-card-item-stub
      icon="M11.19,2.25C10.93,2.25 10.67,2.31 10.42,2.4L3.06,5.45C2.04,5.87 1.55,7.04 1.97,8.05L6.93,20C7.24,20.77 7.97,21.23 8.74,21.25C9,21.25 9.27,21.22 9.53,21.1L16.9,18.05C17.65,17.74 18.11,17 18.13,16.25C18.14,16 18.09,15.71 18,15.45L13,3.5C12.71,2.73 11.97,2.26 11.19,2.25M14.67,2.25L18.12,10.6V4.25A2,2 0 0,0 16.12,2.25M20.13,3.79V12.82L22.56,6.96C22.97,5.94 22.5,4.78 21.47,4.36M11.19,4.22L16.17,16.24L8.78,19.3L3.8,7.29"
      label="給付方式"
      value="定率給付"
    />
    <z-data-card-item-stub
      icon="M11,2V22C5.9,21.5 2,17.2 2,12C2,6.8 5.9,2.5 11,2M13,2V11H22C21.5,6.2 17.8,2.5 13,2M13,13V22C17.7,21.5 21.5,17.8 22,13H13Z"
      label="給付率"
      value="94%"
    >
        未設定の94%を自治体が負担する
    </z-data-card-item-stub>
    <z-data-card-item-stub
      label="端数処理"
      value="切り上げ"
    />
    <z-data-card-item-stub
      icon="M21,6V8H3V6H21M3,18H12V16H3V18M3,13H21V11H3V13Z"
      label="備考"
      value="備考"
    />
    <div
      class="z-data-card-stub-slot-header"
    />
  </div>
  <z-system-meta-card-stub
    createdat="1992-11-15T19:47:52+0900"
    id="10"
    updatedat="1985-11-14T13:24:04+0900"
  />
  <a
    bottom=""
    class="z-fab-stub"
    data-fab=""
    fixed=""
    icon="M20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18,2.9 17.35,2.9 16.96,3.29L15.12,5.12L18.87,8.87M3,17.25V21H6.75L17.81,9.93L14.06,6.18L3,17.25Z"
    nuxt=""
    right=""
    to="/users/1/dws-subsidies/10/edit"
  />
</div>
`;

exports[`pages/users/_id/dws-subsidies/_subsidyId/index.vue z-data-card-item（自治体助成情報） should be rendered z-data-card-item（定率負担） when userDwsSubsidyType is 4 1`] = `
<div
  class="z-data-card-stub"
  data-data-card=""
  title="自治体助成情報"
>
  <div
    class="z-data-card-stub-slot-actions"
  />
  <z-data-card-item-stub
    icon="M19,19H5V8H19M16,1V3H8V1H6V3H5C3.89,3 3,3.89 3,5V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V5C21,3.89 20.1,3 19,3H18V1M17,12H12V17H17V12Z"
    label="適用期間"
    value="10"
  >
    <time
      datetime="1983-05-01"
    >
      昭和58年5月1日
    </time>
    <span>
      〜
    </span>
    <time
      datetime="2020-07-21"
    >
      令和2年7月21日
    </time>
  </z-data-card-item-stub>
  <z-data-card-item-stub
    icon="M15,23H13V21H15V23M19,21H17V23H19V21M15,17H13V19H15V17M7,21H5V23H7V21M7,17H5V19H7V17M19,17H17V19H19V17M15,13H13V15H15V13M19,13H17V15H19V13M21,9A2,2 0 0,1 23,11V23H21V11H11V23H9V15H3V23H1V15A2,2 0 0,1 3,13H9V11A2,2 0 0,1 11,9V7A2,2 0 0,1 13,5H15V1H17V5H19A2,2 0 0,1 21,7V9M19,9V7H13V9H19Z"
    label="助成自治体名"
    value="羽咋市"
  />
  <z-data-card-item-stub
    label="助成自治体番号"
    value="714798"
  />
  <z-data-card-item-stub
    icon="M11.19,2.25C10.93,2.25 10.67,2.31 10.42,2.4L3.06,5.45C2.04,5.87 1.55,7.04 1.97,8.05L6.93,20C7.24,20.77 7.97,21.23 8.74,21.25C9,21.25 9.27,21.22 9.53,21.1L16.9,18.05C17.65,17.74 18.11,17 18.13,16.25C18.14,16 18.09,15.71 18,15.45L13,3.5C12.71,2.73 11.97,2.26 11.19,2.25M14.67,2.25L18.12,10.6V4.25A2,2 0 0,0 16.12,2.25M20.13,3.79V12.82L22.56,6.96C22.97,5.94 22.5,4.78 21.47,4.36M11.19,4.22L16.17,16.24L8.78,19.3L3.8,7.29"
    label="給付方式"
    value="定率負担"
  />
  <z-data-card-item-stub
    icon="M11,2V22C5.9,21.5 2,17.2 2,12C2,6.8 5.9,2.5 11,2M13,2V11H22C21.5,6.2 17.8,2.5 13,2M13,13V22C17.7,21.5 21.5,17.8 22,13H13Z"
    label="負担率"
  >
        利用者負担額を未設定の0%に軽減する
  </z-data-card-item-stub>
  <z-data-card-item-stub
    label="端数処理"
    value="四捨五入"
  />
  <z-data-card-item-stub
    icon="M21,6V8H3V6H21M3,18H12V16H3V18M3,13H21V11H3V13Z"
    label="備考"
    value="備考"
  />
  <div
    class="z-data-card-stub-slot-header"
  />
</div>
`;

exports[`pages/users/_id/dws-subsidies/_subsidyId/index.vue z-data-card-item（自治体助成情報） should be rendered z-data-card-item（本人負担額） when userDwsSubsidyType is 3 1`] = `
<div
  class="z-data-card-stub"
  data-data-card=""
  title="自治体助成情報"
>
  <div
    class="z-data-card-stub-slot-actions"
  />
  <z-data-card-item-stub
    icon="M19,19H5V8H19M16,1V3H8V1H6V3H5C3.89,3 3,3.89 3,5V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V5C21,3.89 20.1,3 19,3H18V1M17,12H12V17H17V12Z"
    label="適用期間"
    value="10"
  >
    <time
      datetime="1983-05-01"
    >
      昭和58年5月1日
    </time>
    <span>
      〜
    </span>
    <time
      datetime="2020-07-21"
    >
      令和2年7月21日
    </time>
  </z-data-card-item-stub>
  <z-data-card-item-stub
    icon="M15,23H13V21H15V23M19,21H17V23H19V21M15,17H13V19H15V17M7,21H5V23H7V21M7,17H5V19H7V17M19,17H17V19H19V17M15,13H13V15H15V13M19,13H17V15H19V13M21,9A2,2 0 0,1 23,11V23H21V11H11V23H9V15H3V23H1V15A2,2 0 0,1 3,13H9V11A2,2 0 0,1 11,9V7A2,2 0 0,1 13,5H15V1H17V5H19A2,2 0 0,1 21,7V9M19,9V7H13V9H19Z"
    label="助成自治体名"
    value="羽咋市"
  />
  <z-data-card-item-stub
    label="助成自治体番号"
    value="714798"
  />
  <z-data-card-item-stub
    icon="M11.19,2.25C10.93,2.25 10.67,2.31 10.42,2.4L3.06,5.45C2.04,5.87 1.55,7.04 1.97,8.05L6.93,20C7.24,20.77 7.97,21.23 8.74,21.25C9,21.25 9.27,21.22 9.53,21.1L16.9,18.05C17.65,17.74 18.11,17 18.13,16.25C18.14,16 18.09,15.71 18,15.45L13,3.5C12.71,2.73 11.97,2.26 11.19,2.25M14.67,2.25L18.12,10.6V4.25A2,2 0 0,0 16.12,2.25M20.13,3.79V12.82L22.56,6.96C22.97,5.94 22.5,4.78 21.47,4.36M11.19,4.22L16.17,16.24L8.78,19.3L3.8,7.29"
    label="給付方式"
    value="定額負担"
  />
  <z-data-card-item-stub
    icon="M13.92 11H18V13H13V15H18V17H13V21H11V17H6V15H11V13H6V11H10.08L5 3H7.37L12 10.29L16.63 3H19L13.92 11Z"
    label="本人負担額"
    value="1,710円"
  />
  <z-data-card-item-stub
    icon="M21,6V8H3V6H21M3,18H12V16H3V18M3,13H21V11H3V13Z"
    label="備考"
    value="備考"
  />
  <div
    class="z-data-card-stub-slot-header"
  />
</div>
`;

exports[`pages/users/_id/dws-subsidies/_subsidyId/index.vue z-data-card-item（自治体助成情報） should be rendered z-data-card-item（給付率） when userDwsSubsidyType is 1 1`] = `
<div
  class="z-data-card-stub"
  data-data-card=""
  title="自治体助成情報"
>
  <div
    class="z-data-card-stub-slot-actions"
  />
  <z-data-card-item-stub
    icon="M19,19H5V8H19M16,1V3H8V1H6V3H5C3.89,3 3,3.89 3,5V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V5C21,3.89 20.1,3 19,3H18V1M17,12H12V17H17V12Z"
    label="適用期間"
    value="10"
  >
    <time
      datetime="1983-05-01"
    >
      昭和58年5月1日
    </time>
    <span>
      〜
    </span>
    <time
      datetime="2020-07-21"
    >
      令和2年7月21日
    </time>
  </z-data-card-item-stub>
  <z-data-card-item-stub
    icon="M15,23H13V21H15V23M19,21H17V23H19V21M15,17H13V19H15V17M7,21H5V23H7V21M7,17H5V19H7V17M19,17H17V19H19V17M15,13H13V15H15V13M19,13H17V15H19V13M21,9A2,2 0 0,1 23,11V23H21V11H11V23H9V15H3V23H1V15A2,2 0 0,1 3,13H9V11A2,2 0 0,1 11,9V7A2,2 0 0,1 13,5H15V1H17V5H19A2,2 0 0,1 21,7V9M19,9V7H13V9H19Z"
    label="助成自治体名"
    value="羽咋市"
  />
  <z-data-card-item-stub
    label="助成自治体番号"
    value="714798"
  />
  <z-data-card-item-stub
    icon="M11.19,2.25C10.93,2.25 10.67,2.31 10.42,2.4L3.06,5.45C2.04,5.87 1.55,7.04 1.97,8.05L6.93,20C7.24,20.77 7.97,21.23 8.74,21.25C9,21.25 9.27,21.22 9.53,21.1L16.9,18.05C17.65,17.74 18.11,17 18.13,16.25C18.14,16 18.09,15.71 18,15.45L13,3.5C12.71,2.73 11.97,2.26 11.19,2.25M14.67,2.25L18.12,10.6V4.25A2,2 0 0,0 16.12,2.25M20.13,3.79V12.82L22.56,6.96C22.97,5.94 22.5,4.78 21.47,4.36M11.19,4.22L16.17,16.24L8.78,19.3L3.8,7.29"
    label="給付方式"
    value="定率給付"
  />
  <z-data-card-item-stub
    icon="M11,2V22C5.9,21.5 2,17.2 2,12C2,6.8 5.9,2.5 11,2M13,2V11H22C21.5,6.2 17.8,2.5 13,2M13,13V22C17.7,21.5 21.5,17.8 22,13H13Z"
    label="給付率"
    value="94%"
  >
        未設定の94%を自治体が負担する
  </z-data-card-item-stub>
  <z-data-card-item-stub
    label="端数処理"
    value="切り上げ"
  />
  <z-data-card-item-stub
    icon="M21,6V8H3V6H21M3,18H12V16H3V18M3,13H21V11H3V13Z"
    label="備考"
    value="備考"
  />
  <div
    class="z-data-card-stub-slot-header"
  />
</div>
`;

exports[`pages/users/_id/dws-subsidies/_subsidyId/index.vue z-data-card-item（自治体助成情報） should be rendered z-data-card-item（給付額） when userDwsSubsidyType is 2 1`] = `
<div
  class="z-data-card-stub"
  data-data-card=""
  title="自治体助成情報"
>
  <div
    class="z-data-card-stub-slot-actions"
  />
  <z-data-card-item-stub
    icon="M19,19H5V8H19M16,1V3H8V1H6V3H5C3.89,3 3,3.89 3,5V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V5C21,3.89 20.1,3 19,3H18V1M17,12H12V17H17V12Z"
    label="適用期間"
    value="10"
  >
    <time
      datetime="1983-05-01"
    >
      昭和58年5月1日
    </time>
    <span>
      〜
    </span>
    <time
      datetime="2020-07-21"
    >
      令和2年7月21日
    </time>
  </z-data-card-item-stub>
  <z-data-card-item-stub
    icon="M15,23H13V21H15V23M19,21H17V23H19V21M15,17H13V19H15V17M7,21H5V23H7V21M7,17H5V19H7V17M19,17H17V19H19V17M15,13H13V15H15V13M19,13H17V15H19V13M21,9A2,2 0 0,1 23,11V23H21V11H11V23H9V15H3V23H1V15A2,2 0 0,1 3,13H9V11A2,2 0 0,1 11,9V7A2,2 0 0,1 13,5H15V1H17V5H19A2,2 0 0,1 21,7V9M19,9V7H13V9H19Z"
    label="助成自治体名"
    value="羽咋市"
  />
  <z-data-card-item-stub
    label="助成自治体番号"
    value="714798"
  />
  <z-data-card-item-stub
    icon="M11.19,2.25C10.93,2.25 10.67,2.31 10.42,2.4L3.06,5.45C2.04,5.87 1.55,7.04 1.97,8.05L6.93,20C7.24,20.77 7.97,21.23 8.74,21.25C9,21.25 9.27,21.22 9.53,21.1L16.9,18.05C17.65,17.74 18.11,17 18.13,16.25C18.14,16 18.09,15.71 18,15.45L13,3.5C12.71,2.73 11.97,2.26 11.19,2.25M14.67,2.25L18.12,10.6V4.25A2,2 0 0,0 16.12,2.25M20.13,3.79V12.82L22.56,6.96C22.97,5.94 22.5,4.78 21.47,4.36M11.19,4.22L16.17,16.24L8.78,19.3L3.8,7.29"
    label="給付方式"
    value="定額給付"
  />
  <z-data-card-item-stub
    icon="M13.92 11H18V13H13V15H18V17H13V21H11V17H6V15H11V13H6V11H10.08L5 3H7.37L12 10.29L16.63 3H19L13.92 11Z"
    label="給付額"
    value="1,710円"
  />
  <z-data-card-item-stub
    icon="M21,6V8H3V6H21M3,18H12V16H3V18M3,13H21V11H3V13Z"
    label="備考"
    value="備考"
  />
  <div
    class="z-data-card-stub-slot-header"
  />
</div>
`;
